src/Featurizers/NormalizeFeaturizer.h

```diff
@@ -0,0 +1,417 @@
+>
+class NormalizeEstimatorImpl : public TransformerEstimator<IteratorRangeT, std::vector<std::double_t>> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using TransformedType                   = std::vector<std::double_t>;
+    using BaseType                          = TransformerEstimator<IteratorRangeT, TransformedType>;
+    using TransformerType                   = NormalizeTransformer<IteratorRangeT>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    NormalizeEstimatorImpl(AnnotationMapsPtr pAllColumnAnnotations, size_t colIndex);
+    ~NormalizeEstimatorImpl(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(NormalizeEstimatorImpl);
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    size_t const                            _colIndex;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+    bool begin_training_impl(void) override;
+    FitResult fit_impl(IteratorRangeT const *, size_t) override;
+    void complete_training_impl(void) override;
+
+    // MSVC has problems when the definition is separate from the declaration
+    typename BaseType::TransformerUniquePtr create_transformer_impl(void) override {
+        // ----------------------------------------------------------------------
+        using VectorNormsAnnotationData = Components::VectorNormsAnnotationData<UpdaterT>;
+        using VectorNormsEstimator       = Components::VectorNormsEstimator<IteratorRangeT, UpdaterT, MaxNumTrainingItemsV>;
+        // ----------------------------------------------------------------------
+
+        VectorNormsAnnotationData const & data(VectorNormsEstimator::get_annotation_data(this->get_column_annotations(), _colIndex, Components::VectorNormsEstimatorName));
+
+        return typename BaseType::TransformerUniquePtr(new NormalizeTransformer<IteratorRangeT>(data.Norms));
+    }
+};
+
+} // namespace Details
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         NormalizeEstimator
+///  \brief         Chains up VectorNormsEstimator and NormalizeTransformerImpl
+///                 to normalize input vector data so that it's norm becoming 1
+///
+template <
+    typename IteratorRangeT,
+    typename UpdaterT,
+    size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()
+>
+class NormalizeEstimator :

```

Comment:
>NormalizeEstimator [](start = 6, length = 18)

Nit: decorate this name to make it clear that it shouldn't be used in isolation; maybe `NormalizeEstimatorImpl`?